controllers/PasswordController/change.js

```diff
@@ -0,0 +1,38 @@
+const User = require('../../models/Users');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+/**
+ * @description recieves a token from the user, verifies the email and sends a frontend link with token to the user's email
+ */
+const change = async (req, res, next) => {
+  const { token , oldPassword, newPassword } = req.body;
+  const { JWT_SECRET } = process.env;
+
+  jwt.verify(token, JWT_SECRET, (err, decoded) => {
+    if (err) {
+      return res.status(401).json({ status: 401, message: 'Invalid token' });
+    }
+    decodedUser = decoded;
+  });
+
+  const user = await User.findOne({ where: { id: decodedUser.id } });
+
+  if (!user) {
+    return res.status(404).json({ message: 'User not found' });
+  }
+  const isMatch = bcrypt.compareSync(oldPassword, user.password);
```

Comment:
prefer async.

Reply:
for only the user or for all the awaits?